### DISCUSSION TITLE
Migrate to the new Dropbox

### DISCUSSION THREAD

**Discussion by joshmedeski** [lra/mackup#1894] [Q&A]
Hi @lra, do you think it's a good idea to create formal documentation for how to migrate to the latest Dropbox? 

Dropbox has changed the location of their files upon an upgrade. If anyone can clarify what that upgrade process looks like that would be helpful.

I've given some instructions on #1858, but if people update their Dropbox and the file location changes it will break things if they don't properly uninstall and reinstall Mackup manually. Or we come up with a script that can migrate existing backups to new directory location if a newer version of Mackup is detected.

No matter what, I think Dropbox users will have to manually get involved to migrate their Mackup backup.

Let me know your thoughts here. Anyone else is welcome to join this discussion to help us come up with a smooth transition for Dropbox users.

**Comment by joshmedeski** [comment]
Some thoughts / questions:

1. If the user already ran the Dropbox update can we programmatically recover the mackup symlinks?
2. Do we need to programmatically support the old and new versions of Dropbox right now, or will `get_dropbox_folder_location` resolve correctly no matter what?

**Comment by erickinsella** [comment]
> If anyone can clarify what that upgrade process looks like that would be helpful.

Dropbox sent out an email stating that there was a new version of the MacOS app earlier this week. 

![2022-12-07 at 11 12 AM](https://user-images.githubusercontent.com/4501362/206246001-f27d64ab-f364-43df-9796-214babb20d3e.png)

The Learn more link went [here.]( https://help.dropbox.com/installs/macos-faqs?_ad=6738_20013347&_camp=p1_bo_co_al_tx_b_Project_Jupiter-FPGA&_tk=email)

The app itself showed a notification that there was an impending update. When clicked on it mentioned briefly that the process would be moving files to a new location and may take some time. Also that the process would be done in the background.

**Comment by anthonysomerset** [comment]
I suppose the best way to do this would be some script or migration command that you would provide the new mackup path to, the script would then read the folder for whats in there, then effectively do a "mackup restore" over the top of the old/broken symlinks as it should be able to reasonably decipher the original location on disk where to update the symlink

not sure how to do this without some form of user prompting/intervention otherwise - one could potentially logically map
`~/Dropbox/<somepath>` to `~/Library/CloudStorage/Dropbox/<somepath>` and then do a mackup restore type run that will just update the symlink based on what it finds in the new directory accordingly but there may still be some edge cases

come to think of it, i don't see why you couldn't just do a mackup restore after updating the config - although .mackup.cfg is also a symlink so might need to manage that chicken/egg scenario manually

**Comment by commadelimited** [comment]
I was just bitten with this. :(

All my files appear to still be in Dropbox, but Mackup doesn't seem to know about them. I reinstalled Mackup and now I'd just like to relink my Dropbox folder. I thought running mackup restore would work, as if I was just setting up a new machine. But it appears that the new Dropbox path is `/Users/<username>/Library/CloudStorage/Dropbox` instead of `/Users/<username>/Dropbox/Mackup/`.

Is there a way to tell Mackup the new location for my files?

**Reply by commadelimited** [reply]
For anyone else who comes across this, I was able to fix the issue by using the commands suggested in this comment:
https://github.com/lra/mackup/issues/1386#issuecomment-499193594
